src/components/pages/dashboard-pages/Students.jsx

```diff
@@ -1,11 +1,327 @@
+import { useEffect, useState } from 'react';
+import Loader from '../../common/Loader';
+import {
+  getAllUsersInfo,
+  getAllStudentLessons,
+  getClassesData,
+} from '../../../util/DataBaseRequests';
+import { useAuth } from '../../../AuthProvider';
+import { useNavigate } from 'react-router-dom';
+import {
+  calculateAge,
+  formatDateWithoutWeekday,
+} from '../../../util/NotificationsUtils';
+import { formatDateWithWeekday } from '../../../util/NotificationsUtils';
+
 const TeacherStudents = () => {
+  const navigate = useNavigate();
+  const { userData } = useAuth();
+  const [students, setStudents] = useState([]);
+  const [selectedId, setSelectedId] = useState();
+  const [selectedStudent, setSelectedStudent] = useState();
+  const [studentsError, setStudentsError] = useState({});
+  const [isLoading, setIsLoading] = useState(false);
+  const [studentLessons, setStudentLessons] = useState([]);
+  const [classTitles, setClassTitles] = useState([]);
+
+  useEffect(() => {
+    setIsLoading(true);
+    const getTeacherStudents = async () => {
+      const token = userData.token;
+      try {
+        const allUsers = await getAllUsersInfo(token);
+        const myStudents = userData.myStudents;
+        const filteredStudents = allUsers.filter((user) =>
+          myStudents.includes(user._id)
+        );
+        if (filteredStudents.length === 0) {
+          setIsLoading(false);
+          setStudentsError({
+            noStudentsError: `You haven't added any students yet.`,
```

Comment:
I am not sure, but for me this error message looks like the teacher could add student without application.

Reply:
But we have below button with test "check your applications" that navigate you to applications>

Example - screenshot in pr description.

Reply:
Thats true...